### DISCUSSION TITLE
get-dbaxesession | read-dbaxefile gives access issue

### DISCUSSION THREAD

**Discussion by RoyvanDalen** [dataplat/dbatools#8663] [Q&A]
I did not want to create a issue for this, as mentioned in dbatools documentation it states that many issues are caused by access permissions configuration, I assume we are dealing with security setup in this case too. We want to use the get-dbaxesession | read-dbaxefile to get extended events data into tables, because this handles the xml data very nicely compared to other solutions. But on almost all our servers we get "warning [date][read-dbaxefile] c:\MSSQL\log\system_health.xel cannot be accessed from serverA 
Does account DomainB\AccountC have access?"
We are running dbatools 1.1.142 and have PS 5 on Windows 2019. It does work on 2 servers, but i tested like 10 other servers where it gives us this error. If i change it to "Get-ChildItem C:\temp\xe\*.xel | Read-DbaXEFile"  it does work, we checked file and share settings but see no difference between these servers. Any idea where to look for what can cause this? it seems specific to get-dbaexsession which uses unc ?  thanks for assistance

**Comment by wsmelton** [comment]
I believe that command generates an admin share UNC path for the files output by Get-DbaXESession. The service account used by ServerB will need admin access to ServerA.

**Comment by RoyvanDalen** [comment]
Hi Shawn, yes we know you need admin permissions, we checked and all looks the same. That is what makes it so strange , but we have a work around getting the files without get-dbaxesession so we are not using get-dbaxesession anymore. Ill double check the admin access but I am sure these service accounts are all local admin on their servers

**Reply by wsmelton** [reply]
Miss typed, updated above response. Your service account needs access to the other server in order to access the admin share of another server.

**Comment by potatoqualitee** [comment]
Hello, can you please run the the following command:
```
Install-Module dbatools -AllowPrerelease -Scope CurrentUser
```

then restart powershell, and try again. 

this is prerelease2 of dbatools 2.0, which contains a ton of fixes
